packages/devtools-client-adapters/src/types.js

```diff
@@ -0,0 +1,162 @@
+// @flow
+/**
+ * These are intended to be generic types used by Firefox, Chrome, Node, and
+ * possibly others.
+ *
+ * They can be imported into clients by using the following Flow import:
+ * `import type { Location } from 'devtools-client-adapters'`
+ */
+
+/**
+ * Breakpoint ID
+ *
+ * @memberof types
+ * @static
+ */
+export type BreakpointId = string;
+
+/**
+ * Source ID
+ *
+ * @memberof types
+ * @static
+ */
+export type SourceId = string;
+
+/**
+ * Actor ID
+ *
+ * @memberof types
+ * @static
+ */
+export type ActorId = string;
+
+/**
+ * Source File Location
+ *
+ * @memberof types
+ * @static
+ */
+export type Location = {
+  sourceId: SourceId,
+  line: number,
+  column?: number
+};
+
+/**
+ * Breakpoint
+ *
+ * @memberof types
+ * @static
+ */
+export type Breakpoint = {
+  id: BreakpointId,
+  location: Location,
+  loading: boolean,
+  disabled: boolean,
+  text: string,
+  condition: ?string,
+};
+
+/**
+ * Breakpoint Result is the return from an add/modify Breakpoint request
+ *
+ * @memberof types
+ * @static
+ */
+export type BreakpointResult = {
+  id: ActorId,
+  actualLocation: Location
+};
+
+/**
+ * Source
+ *
+ * @memberof types
+ * @static
+ */
+export type Source = {
+  id: SourceId,
+  url?: string,
+  sourceMapURL?: string
+};
+
+/**
+ * Frame ID
+ *
+ * @memberof types
+ * @static
+ */
+export type FrameId = string;
+
+/**
+ * Frame
+ * @memberof types
+ * @static
+ */
+export type Frame = {
+  id: FrameId,
+  displayName: string,
+  location: Location,
+  source?: Source,
+  scope: Scope,
+  // FIXME Define this type more clearly
+  this: Object
+};
+
+/**
+ * Why is the Debugger Paused?
+ * This is the generic state handling the reason the debugger is paused.
+ * Reasons are usually related to "breakpoint" or "debuggerStatement"
+ * and should eventually be specified here as an enum.  For now we will
+ * just offer it as a string.
+ * @memberof types
+ * @static
+ */
+export type WhyPaused = {
```

Comment:
These kinds of types will help us get to a more generic version of a feature like this.